Kernel.cc

```diff
@@ -1586,13 +1598,14 @@ int Kernel::unlink(char *pathname) {
 
       	// Get the full path of pathname
 	char *fullPath = getFullPath(pathname);
```

Comment:
Literally move this line down by 1 and it works, with some other changes :1st_place_medal: :balloon: . Calling `getFullPath` twice (open also calls the function), will corrupt their result because they use a static variable for the return value.